pkg/controller/machine/machine_controller.go

```diff
@@ -274,15 +278,33 @@ func (r *ReconcileMachine) isDeleteAllowed(machine *clusterv1.Machine) bool {
 	return node.UID != machine.Status.NodeRef.UID
 }
 
-func (r *ReconcileMachine) deleteNode(ctx context.Context, name string) error {
-	var node corev1.Node
-	if err := r.Client.Get(ctx, client.ObjectKey{Name: name}, &node); err != nil {
-		if apierrors.IsNotFound(err) {
-			klog.V(2).Infof("Node %q not found", name)
-			return nil
+func (r *ReconcileMachine) deleteNode(ctx context.Context, cluster *clusterv1.Cluster, name string) error {
+	if cluster == nil {
+		var node corev1.Node
+		if err := r.Client.Get(ctx, client.ObjectKey{Name: name}, &node); err != nil {
+			if apierrors.IsNotFound(err) {
+				klog.V(2).Infof("Node %q not found", name)
+				return nil
+			}
+			klog.Errorf("Failed to get node %q: %v", name, err)
+			return err
 		}
-		klog.Errorf("Failed to get node %q: %v", name, err)
-		return err
+		return r.Client.Delete(ctx, &node)
+	}
+
+	remoteClient, err := remote.NewClusterClient(r.Client, cluster)
+	if err != nil {
+		klog.Errorf("Error creating a remote client for cluster %q while deleting Machine %q, won't retry: %v",
+			cluster.Name, name, err)
+		return nil
 	}
-	return r.Client.Delete(ctx, &node)
+
+	corev1Remote, err := remoteClient.CoreV1()
+	if err != nil {
+		klog.Errorf("Error creating a remote client for cluster %q while deleting Machine %q, won't retry: %v",
+			cluster.Name, name, err)
+		return nil
+	}
+
+	return corev1Remote.Nodes().Delete(name, &metav1.DeleteOptions{})
```

Comment:
This has potential to loop forever if the node doesn't exist already. This should also error check that if the object can't be found then it's a success, not a retryable error. Unless `.Delete` does this automatically...:rocket:

Reply:
The caller does the check at the moment, `!apierrors.IsNotFound(err)`. Is this what you were referring to?

Reply:
yep! thanks